cmd/snap-bootstrap/cmd_initramfs_mounts.go

```diff
@@ -924,23 +951,14 @@ func (m *recoverModeStateMachine) unlockSaveFallbackKey() (stateFunc, error) {
 	// the user to enter, and what we are unlocking (as currently the prompt
 	// says "recovery key" and the partition UUID for what is being unlocked)
 	unlockRes, unlockErr := secbootUnlockVolumeUsingSealedKeyIfEncrypted(m.disk, "ubuntu-save", saveFallbackKey, unlockOpts)
-	const partitionOptionalIfUnencrypted = true
-	if err := m.setUnlockStateWithFallbackKey("ubuntu-save", unlockRes, unlockErr, partitionOptionalIfUnencrypted); err != nil {
+	if err := m.setUnlockStateWithFallbackKey("ubuntu-save", unlockRes, unlockErr); err != nil {
 		return nil, err
 	}
 	if unlockErr != nil {
 		// all done, nothing left to try and mount, mounting ubuntu-save is the
 		// last step but we couldn't find or unlock it
 		return nil, nil
 	}
-
-	// do a consistency check to make sure that if we found ubuntu-data
-	// unencrypted that we don't also mount ubuntu-save as encrypted
-	data := m.degradedState.partition("ubuntu-data")
-	if unlockRes.IsEncrypted && data.FindState == partitionFound && !assumeEncrypted {
-		return nil, fmt.Errorf("inconsistent encryption status for disk %s: ubuntu-data (device %s) was found unencrypted but ubuntu-save (device %s) was found to be encrypted", m.disk.Dev(), data.fsDevice, unlockRes.FsDevice)
-	}
```

Comment:
I really like that we eliminated this situation entirely from the state diagram by adding more states/transitioning differently, thanks for this refactor

Reply:
also if I'm reading the code correctly, now if there is an unencrypted ubuntu-data, then we search for specifically unencrypted ubuntu-save, and if we didn't find it (because i.e. there is an encrypted ubuntu-save-enc), then we just pass along not attempting to mount it at all